UI/obs-app.hpp

```diff
@@ -154,7 +164,11 @@ private slots:
 
 	inline QMainWindow *GetMainWindow() const { return mainWindow.data(); }
 
-	inline config_t *GlobalConfig() const { return globalConfig; }
+	inline config_t *GetAppConfig() const { return appConfig; }
+	inline config_t *GetUserConfig() const { return userConfig; }
+	std::string userConfigLocation;
+	std::string userScenesLocation;
+	std::string userProfilesLocation;
```

Comment:
Is there any particular reason not to use `std::filesystem::path` for these?

Reply:
Mainly because they are treated as strings throughout most of the codebase. In theory one could keep them as filesystem paths and always convert into `u8string` and into `c_str` from those, but that would create unnecessary overhead as those variables are more often used for string composition than actual filesystem operations (in contrast to the path objects used in the `OBSSceneCollection` and `OBSProfile` objects).

Reply:
Just from a brief look through the diff it seems that those aren't used in a ton of places, and in the majority of them they are converted to `std::filesystem::path` anyway. So it seems to me that storing them as such directly would be simpler. This may also need adjustments of some helper functions like `GetClosestUnusedFileName()` to support it, but that should be trivial (in that specific example the use of libobs `os_` APIs could also be substituted with more `std::filesystem` features). Maybe also add an overload to the `ConfigFile` wrapper's `Open()` method that takes a reference to an `std::filesystem::path` object.

Reply:
Those APIs are precisely the reason why they are strings - all these APIs are involved:

* `ConfigFile.open()` (`std::string`) 
* `GetClosestUnusedFileName()` (`std::string`)
* `GetFileSafeName()` (`char *` and `std::string`)
* `os_quick_write_utf8_file` (`char *`)

So in general I opted to keep the strings as such for as long as possible to minimise the impact on existing API functions and convert to paths only when actual file system interaction takes place.

The necessity to either explicitly convert them into a UTF-8 type (or use deprecated UTF-8-specific functions) seems to makes it even more prudent to me to keep using strings for as long as possible and not use `std::filesystem::path` as storage.